### PR TITLE
Add account security section to set 2FA methods

### PR DESCRIPTION
**Pull request by scruti**
[Jira ticket](https://regulatorydelivery.atlassian.net/browse/COSBETA-1135)

This PR adds a section with security preferences for secondary authentication into the user's `my_account` page.

From the account security section, users will be able to set or update both:
- Text authentication (Direct OTP)
- App authentication (Time based OTP)

### Detailed changes:
- Turned the "mobile number change" page into secondary authentication SMS setup.
- Both sms/text setup and app setup pages require:
  - To have successfully passed the 2FA check in the last 5 minutes.
  - To introduce user password in order to validate changes.
- A confirmation message gets displayed when redirected to the user account page after successfully setting/updating the authentication methods.
- When changes are done to either mobile number or totp_secret_key, secondary authentication methods for the user are updated.

### Screenshots

---
**When a method is not set it offers a link to set it up:**

---
**The Authenticator App setup page needs password confirmation like mobile number one:**

### CODE REVIEW

File: cosmetics-web/.rubocop.yml [hunk 1]
```diff
@@ -70,14 +70,12 @@ Metrics/BlockLength:
 
 RSpec/ExpectInHook:
   Exclude:
-    - spec/features/account/change_mobile_number_spec.rb
```
Comment (scruti) [inline cosmetics-web/.rubocop.yml:73]: No longer needed

File: cosmetics-web/.rubocop.yml [hunk 2]
```diff
@@ -70,14 +70,12 @@ Metrics/BlockLength:
...
     - spec/features/account/change_name_spec.rb
     - spec/features/account/change_password_spec.rb
     - spec/features/account/change_email_spec.rb
 
 RSpec/LetSetup:
   Exclude:
-    - spec/features/account/change_mobile_number_spec.rb
```
Comment (scruti) [inline cosmetics-web/.rubocop.yml:80]: No longer needed

File: cosmetics-web/.rubocop.yml [hunk 3]
```diff
@@ -90,10 +88,6 @@ RSpec/MultipleExpectations:
 RSpec/NestedGroups:
   Max: 6
 
-RSpec/ScatteredSetup:
-  Exclude:
-    - spec/features/account/change_mobile_number_spec.rb
-
```
Comment (scruti) [inline cosmetics-web/.rubocop.yml:96]: No longer needed

File: cosmetics-web/app/controllers/secondary_authentication/sms/setup_controller.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+module SecondaryAuthentication
+  module Sms
+    class SetupController < SubmitApplicationController
```
Comment (scruti) [inline cosmetics-web/app/controllers/secondary_authentication/sms/setup_controller.rb:3]: This used to be the `MobileNumberController` that has been adapted for the secondary authentication setup domain.

File: cosmetics-web/app/forms/registration/account_security_form.rb [hunk 5]
```diff
@@ -49,7 +49,6 @@ def update!
         confirmed_at: Time.zone.now,
         last_totp_at: secondary_authentication.last_totp_at,
         totp_secret_key: (secret_key if app_authentication_selected?),
-        secondary_authentication_methods: secondary_authentication_methods,
```
Comment (scruti) [inline cosmetics-web/app/forms/registration/account_security_form.rb:52]: This now gets internally updated in `User` class.

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on review app , looking good as expected.

File: cosmetics-web/app/forms/secondary_authentication/app/auth_form.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                if: -> { INTEGER_REGEX.match?(otp_code) }
+      validate :correct_otp_validation
+
+      delegate :last_totp_at, to: :secondary_authentication
+
+      def otp_code=(code)
+        super(code.to_s.strip)
+      end
+
+      def back_link?
```
Comment (slawosz) [inline cosmetics-web/app/forms/secondary_authentication/app/auth_form.rb:32]: I don't think this is best practice to have logic that decides on frontend in form.

**Review by slawosz — APPROVED** [review]

File: cosmetics-web/app/forms/secondary_authentication/app/auth_form.rb (hunk 6 above)
Comment (scruti) [reply cosmetics-web/app/forms/secondary_authentication/app/auth_form.rb:32]: Good point. Going to extract this and the one in the SMS form.

**Pull request merged**